### PR TITLE
Use httpc and Don't Retry Failed Requests

### PR DESCRIPTION
**Pull request by JEG2**
Hopefully this is better than filling up a hackney pool.

### CODE REVIEW

File: .github/workflows/chromedriver.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     strategy:
       matrix:
         otp: [21.x]
-        elixir: [1.7.x, 1.8.x, 1.9.x]
+        elixir: [1.7.x, 1.8.x, 1.9.x, 1.10.x, 1.11.x]
```
Comment (JEG2) [inline .github/workflows/chromedriver.yml:17]: I didn't modify this file.  How did this happen???

File: lib/wallaby/chrome/chromedriver/readiness_checker.ex [hunk 2]
```diff
@@ -38,7 +38,9 @@ defmodule Wallaby.Chrome.Chromedriver.ReadinessChecker do
     )
   end
 
-  @default_httpoison_options [hackney: [pool: :wallaby_pool]]
+  # The following is only used for a status check in an external library.
+  # Our code uses `httpc` instead of `hackney`.
+  @default_httpoison_options [hackney: []]
```
Comment (JEG2) [inline lib/wallaby/chrome/chromedriver/readiness_checker.ex:43]: This is silly.  We could do our own check, without the dependency.

Comment (JEG2) [reply lib/wallaby/chrome/chromedriver/readiness_checker.ex:43]: PIO doesn't use this code.

File: lib/wallaby/httpclient.ex [hunk 3]
```diff
@@ -37,47 +43,59 @@ defmodule Wallaby.HTTPClient do
     make_request(method, url, Jason.encode!(params))
   end
 
-  defp make_request(method, url, body), do: make_request(method, url, body, 0, [])
-
-  @spec make_request(method, url, String.t() | map, non_neg_integer(), [String.t()]) ::
+  @spec make_request(method, url, String.t() | map) ::
```
Comment (JEG2) [inline lib/wallaby/httpclient.ex:46]: 🥩 Removing retries and switching to httpc.

File: mix.exs [hunk 4]
```diff
@@ -41,7 +41,7 @@ defmodule Wallaby.Mixfile do
   end
 
   def application do
-    [extra_applications: [:logger], mod: {Wallaby, []}]
+    [extra_applications: [:logger, :inets, :ssl], mod: {Wallaby, []}]
```
Comment (JEG2) [inline mix.exs:44]: 🥩 httpc requirements.

File: test/wallaby/session_store_test.exs [hunk 5]
```diff
@@ -8,7 +8,7 @@ defmodule Wallaby.SessionStoreTest do
       assert [] = SessionStore.list_sessions_for()
       {:ok, session} = Wallaby.driver().start_session([])
       :ok = SessionStore.monitor(session)
-      assert [session] = SessionStore.list_sessions_for()
+      assert [_session] = SessionStore.list_sessions_for()
```
Comment (JEG2) [inline test/wallaby/session_store_test.exs:11]: Silencing a warning.

File: mix.exs [hunk 6]
```diff
@@ -52,8 +52,6 @@ defmodule Wallaby.Mixfile do
   defp deps do
     [
       {:jason, "~> 1.1"},
-      {:hackney, "~> 1.16.0"},
-      {:httpoison, "~> 1.7.0"},
```
Comment (hazen) [inline mix.exs:56]: @JEG2 Prolly want to regen or update the lock file. I see several deps using `hackney` but none using `httpoison`

**Review by hazen — APPROVED** [review]

**Pull request merged**